### PR TITLE
lint: fix shellcheck for misc/systemv/nix-daemon

### PR DESCRIPTION
**Pull request by fzakaria**
# Motivation

Got shellcheck passing for misc/systemv/nix-daemon

Not sure how to test this since it's not running on my NixOS machine and I see no references to it in the directory otherwise.

# Context
See #10795

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -324,6 +324,7 @@
             ++ pkgs.nixComponents.nix-external-api-docs.nativeBuildInputs
             ++ [
               pkgs.buildPackages.cmake
+              pkgs.shellcheck
```
Comment (fzakaria) [inline flake.nix:327]: I wanted to run shellcheck myself manually.

Shellcheck has a diff output style that you can try to apply for auto fixes for lots of cases now
```shellcheck misc/systemv/nix-daemon -f diff | git apply```

File: misc/systemv/nix-daemon [hunk 2]
```diff
@@ -34,21 +34,28 @@ else
...
 # Source function library.
+# shellcheck source=/dev/null
 . /etc/init.d/functions
 
 LOCKFILE=/var/lock/subsys/nix-daemon
 RUNDIR=/var/run/nix
 PIDFILE=${RUNDIR}/nix-daemon.pid
 RETVAL=0
 
-base=${0##*/}
```
Comment (fzakaria) [inline misc/systemv/nix-daemon:44]: warning from shellcheck that this has no reference.

**Comment by roberth** [comment]
> Not sure how to test this since it's not running on my NixOS machine and I see no references to it in the directory otherwise.

Oof. If we don't have the tests to maintain this file, maybe we shouldn't pretend to.

Or would this be used in, say, NixBSD? Do they have a test infra we could reuse?

**Comment by fzakaria** [comment]
Cool -- in the meantime i'll pickup a different script.
We can merge when we hear back the answer.

**Comment by roberth** [comment]
Not used there.
I'm not sure how to proceed with this. It could be an improvement, but making a change without testing it also poses a risk.
Should we remove it?

**Pull request merged**